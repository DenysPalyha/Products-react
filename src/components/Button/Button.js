import React from "react";
import styles from "./Button.module.scss";
import PropTypes from "prop-types";

const Button = ({ bgColor, textBtn, hendlerClik, disabledBtn }) => {
  return (
    <button
      type="button"
      style={{ backgroundColor: bgColor }}
      disabled={disabledBtn}
      className={styles.btn}
      onClick={hendlerClik}
    >
      {textBtn}
    </button>
  );
};

Button.propTypes = {
  bgColor: PropTypes.string,
  textBtn: PropTypes.node,
  hendlerClik: PropTypes.func,
  disabledBtn: PropTypes.bool,
};

Button.defaultProps = {
  bgColor: "",
  textBtn: "",
  disabledBtn: false,
  hendlerClikOpenModal: () => {},
};

export default Button;
