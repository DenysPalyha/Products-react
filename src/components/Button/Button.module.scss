.btn {
  font-size: 14px;
  line-height: 1;
  margin: 0;
  padding: 10px;
  outline: none;
  border: 0;
  border-radius: 5px;
  background-color: transparent;
  cursor: pointer;
  color: #fff;
}
