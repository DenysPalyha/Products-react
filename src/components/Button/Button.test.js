import { render, screen, fireEvent } from "@testing-library/react";

import Button from "./Button";

const handleClick = jest.fn();

describe("Render Button", () => {
  test("should Button render", () => {
    const { asFragment } = render(
      <Button bgColor="#fff" textBtn="Testing Button" disabledBtn={false} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Button hendler click", () => {
  test("should Button hendler click worcking", () => {
    render(
      <Button
        bgColor="#fff"
        textBtn="Testing Button"
        hendlerClik={handleClick}
        disabledBtn={false}
      />
    );

    const btn = screen.getByText("Testing Button");
    fireEvent.click(btn);
    expect(handleClick).toHaveBeenCalled();
  });
});
