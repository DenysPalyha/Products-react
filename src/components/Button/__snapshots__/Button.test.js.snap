// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button should Button render 1`] = `
<DocumentFragment>
  <button
    class="btn"
    style="background-color: rgb(255, 255, 255);"
    type="button"
  >
    Testing Button
  </button>
</DocumentFragment>
`;
